cpp_exercise/src/lib/ipcpipelib.cc

```diff
@@ -84,12 +96,11 @@ PipeReceiver::PipeReceiver(const std::string & filename): _file_name(filename){
         std::cout<<"**************************************************************"<<std::endl;
         std::cout<<"**************************************************************"<<std::endl;
         std::cout<<"          File Name: "<<_file_name<<std::endl;
-        remove(_myfifo);
+        remove(pipe_fifo_name.c_str());
```

Comment:
Tip: the issue by removing the pipe if one was already created is that if the sender was connected to this pipe it will wait for the receiver.

Reply:
If a pipe with the same name is already created and you launch the sender first, it will connect to the existing pipe. And the receiver will unlink it and create a new one. They won't be able to communicate between them because there are in two different pipes.